### DISCUSSION TITLE
Table: allow to pass functions to `field` properties

### DISCUSSION THREAD

**Discussion by tsinevik** [primefaces/community#1699] [PrimeNG] [answered]
As the title says I believe it would be great if the table and all related components would allow to pass functions to `field` properties, e.g. for `globalFilterFields`, `sortField`, `pSortableColumn`, `p-columnFilter`, etc.

It's easy to display in the table some computed value that is based on multiple properties - all thanks to the template, it provides a great flexibility. But sometimes we also want to filter table using these computed values. But here comes the problem - it's not clear how to use computed value during the filtering or sorting because we can only provide property name to the `field` that contains some fixed value. I guess it's possible to map data to some view model and it might become very cumbersome to do, because now we need to maintain original object and view model. In some cases we may want to sort, filter and display the same column using different values.

I believe allowing to pass function to these `field` will greatly improve experience and flexibility of how the developers use the table.

Technically table already provides this functionality for most `field` properties - although they are typed as `string` or `string[]` it's still possible to pass function and that function would be evaluated correctly thanks to `ObjectUtils.resolveFieldData` function that can extract value from the object using either string, or string with dot notation, or function. But for example `p-columnFilter` doesn't work with functions since it uses provided field as object key, so the function would be converted to string - so no way to use it anymore.

One possible solution may be to convert Table `filters` property to `Map` since it allows to have functions as a key.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks for the feedback, we'll planning to implement this in the future (v18).
